src/inc/piece/state/tetris_piece_state.h

```diff
@@ -0,0 +1,39 @@
+#ifndef JARED_ALL_TETRIS_TETRIS_PIECE_STATE
+#define JARED_ALL_TETRIS_TETRIS_PIECE_STATE
+
+#include "tetris_piece.h"
+
+class TetrisPieceState
+{
+  
+public:
+
+  TetrisPieceState( TetrisPiece& param_piece )
+    : piece( param_piece )
+  {
+  }
+
+  virtual ~TetrisPieceState()
+  {
+  }
+
+  virtual std::unique_ptr<TetrisPieceState> update() = 0;
+
+  virtual std::unique_ptr<TetrisPieceState> determine_next_state() = 0;
```

Comment:
Need this method to handle getting the next state for each piece independently of updating the piece. This allows for checking for input to be handled without advancing the game.